src/gsy_e/models/market/two_sided.py

```diff
@@ -79,6 +79,22 @@ def get_bids(self) -> Dict:
         """
         return deepcopy(self.bids)
 
+    def _update_requirements_prices(self, bid):
+        requirements = []
+        for requirement in bid.requirements or []:
+            updated_requirement = {**requirement}
+            if "price" in updated_requirement:
+                if "original_price" not in updated_requirement:
+                    updated_requirement["original_price"] = updated_requirement["price"]
+
+                energy = updated_requirement.get("energy") or bid.energy
+                updated_price = (self.fee_class.update_incoming_bid_with_fee(
+                    updated_requirement["price"] / energy,
+                    updated_requirement["original_price"] / energy)) * energy
```

Comment:
Nit: given what the `update_incoming_bid_with_fee` method is doing (just selecting one of the two rates) these lines seem a bit redundant. They're basically doing:
```python
updated_price = updated_requirement["price"] or updated_requirement["original_price"]
```
This could be ok if the implementation of `update_incoming_bid_with_fee()` might change in the future. Regardless, these lines seem a bit redundant somehow and, since you wrote the `BaseClassGridFees` interface, you might be the right person to see if this can be changed or if it's better to keep it as it is.

Reply:
One more thing: `GridFees.update_incoming_bid_with_fee()` is not doing what the name implies. It is not updating anything, it's just selecting which rate to apply.

Reply:
First thing popping in my head: given the name, I would probably expect `update_incoming_bid_with_fee()` to take a `Bid` as an argument and modify it. This would internally do what you're doing in `_update_requirements_prices()`. Just an idea, you might find other better ones.

Reply:
@fievelk I agree with your proposals and indeed these modules need to be further refactored, actually I did not write them I only fixed some stylings warnings in this [PR](https://github.com/gridsingularity/gsy-e/pull/1373).
After an offline discussion with Spyros, we agreed not to alter the logic to avoid introducing bugs, especially because we'll revisit and reimplement the entire fees modules after getting rid of the hierarchical setups we currently support (One more point to discuss in the upcoming workshop I assume)